ui/src/app/machines/views/MachineList/PowerColumn/PowerColumn.test.js

```diff
@@ -0,0 +1,96 @@
+import { MemoryRouter } from "react-router-dom";
+import { mount } from "enzyme";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+import React from "react";
+
+import PowerColumn from "./PowerColumn";
+
+const mockStore = configureStore();
+
+describe("PowerColumn", () => {
+  let state;
+  beforeEach(() => {
+    state = {
+      config: {
+        items: []
+      },
+      machine: {
+        errors: {},
+        loading: false,
+        loaded: true,
+        items: [
+          {
+            system_id: "abc123",
+            power_state: "on",
+            power_type: "virsh"
+          }
+        ]
+      }
+    };
+  });
+
+  it("renders", () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <PowerColumn systemId="abc123" />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    expect(wrapper.find("PowerColumn")).toMatchSnapshot();
+  });
+
+  it("displays a Loader if machine in power transition", () => {
+    state.machine.items[0].powerTransition = true;
+
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <PowerColumn systemId="abc123" />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    expect(wrapper.find("Loader").exists()).toBe(true);
+  });
+
+  it("displays the correct power state", () => {
+    state.machine.items[0].power_state = "off";
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <PowerColumn systemId="abc123" />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    expect(wrapper.find('[data-test="power_state"]').text()).toEqual("off");
+  });
+
+  it("displays the correct power type", () => {
```

Comment:
Do you need both tests here? They will both fail it something goes wrong right?

Reply:
I think a good testing principal is tests should try to isolate a single failure, so should generally just have a single assertion.